File: client/src/models/Account.js

export class Profile{
  constructor(data){
    this.id = data.id
    this.name = data.name
    this.picture = data.picture
  }
}

export class CultMemberProfile extends Profile{
  constructor(data){
    super(data)
    this.cultMemberId = data.cultMemberId
  }
}

export class Account extends Profile {
  constructor(data) {
    super(data) // invokes the contractor from the extend class
    // this.id = data.id
    this.email = data.email
    this.phoneNumber = data.phoneNumber
    this.creditCardNumber = data.creditCardNumber
    // this.name = data.name
    // this.picture = data.picture
    // TODO add additional properties if needed
  }
}

File: server/Models/Account.cs
namespace instaCult.Models;


public class Profile : RepoItem<string>
{
  public string Name { get; set; }
  public string Picture { get; set; }
}

public class Account : Profile
{
  public string Email { get; set; }
  public string PhoneNumber { get; set; } 
  public string  CreditCardNumber { get; set; }
}

public class CultMemberProfile : Profile
{
  public int CultMemberId { get; set; }
}